.github/workflows/release.yml

```diff
@@ -61,17 +57,151 @@ jobs:
+            if [[ git ls-remote --exit-code --heads origin ${branch_name} ]]; then
+              # The release branch already exists, we need to make sure that our commit is its current tip
+              branch_head_sha=$(git rev-parse --verify --short refs/heads/patches)
+              if [[ branch_head_sha != commit_sha ]]; then
+                echo "The release branch - ${branch_name} - already exists. ${commit_sha}, the commit you chose when "
+                echo "launching this release, is not its current HEAD (${branch_head_sha}). This is not allowed: you "
+                echo "MUST release from the HEAD of the release branch if it already exists."
+                exit 1
+              fi
+            else
+              # The release branch does not exist.
+              # We need to make sure that the commit SHA that we are releasing is on `main`.
+              git fetch origin main
+              if [[ git branch --contains ${commit_sha} | grep main ]]; then
+                # We can then create create the release branch and set the current commit as its tip
+                git checkout -b ${branch_name} 
+                git push origin ${branch_name}
+              else
+                echo "You must choose a commit from main to create a new release series!"
+                exit 1
+              fi
+            fi
+          fi
+
+  upgrade-gradle-properties:
+    name: Upgrade gradle.properties
+    if: inputs.dry_run == false
+    needs:
+      - get-or-create-release-branch
+    outputs:
+      commit_sha: ${{ steps.gradle-push.outputs.commit_sha }}
+      release_branch: ${{ needs.get-or-create-release-branch.outputs.release_branch }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.release_branch }}
+          token: ${{ secrets.RELEASE_AUTOMATION_BOT_PAT }}
+      - name: Upgrade gradle.properties
+        uses: ./smithy-rs/.github/actions/docker-build
+        with:
+          action: upgrade-gradle-smithy-rs-release ${{ inputs.semantic_version }}
+      - name: Push gradle.properties changes
+        id: gradle-push
+        shell: bash
+        env:
+          SEMANTIC_VERSION: ${{ inputs.semantic_version }}
+          DRY_RUN: ${{ inputs.dry_run }}
+        run: |
+          if [[ git diff-index --quiet HEAD ]]; then
+            # The file was actually changed, we need to commit and push the changes
+            git commit gradle.properties -m "Upgrade the smithy-rs runtime crates version to ${SEMANTIC_VERSION}"
+            echo "Pushing upgraded gradle.properties commits..."
+            git push origin
+          fi
+          echo "commit_sha=$(git rev-parse --verify --short HEAD)" > $GITHUB_OUTPUT
+
+  # We'll need to build a base image again if the previous job upgraded the gradle.properties file.
+  # This job will be a no-op if the version in gradle.properties had already been upgraded.
+  acquire-base-image-2:
```

Comment:
What in `tools/` uses the `gradle.properties` file? Doesn't seem like this should be necessary. Even if a tool reads `gradle.properties` from the filesystem, it shouldn't need an updated Docker image since that version number isn't being compiled into the tool.

Separately, you will probably want an escape hatch for using the latest tools in `main` for the actual release workflow since you may need an updated `publisher` to actually successfully publish.

Reply:
> Separately, you will probably want an escape hatch for using the latest tools in main for the actual release workflow since you may need an updated publisher to actually successfully publish.

My assumption is that you can always merge the upgraded tools into the release branch if that's required, indeed as an escape hatch.